### DISCUSSION TITLE
How to disable animations?

### DISCUSSION THREAD

**Discussion by SnippyValson** [Live-Charts/LiveCharts2#196] [General]
I am plotting some live data and animations sometimes causes distractions. Is it possible to disable animations? I tied setting the animation speed to 0, setting the easing function as Liner etc... None of it worked. Any help would be appreciated.
PS: I'm using cartesian chart.
Thank you.

**Comment by beto-rodriguez** [comment]
Set the `EasingFunction` property to null.

https://github.com/beto-rodriguez/LiveCharts2/blob/master/docs/overview/1.9.animations.md#disable-animations

**Reply by alex4998** [reply]
Today I got to this question of disabling animation and I tried many things including setting EasingFunction to null on different levels: on the CartesianChart declaration in XAML, on creating and initializing Axis. Nothing worked. Anyway, animation kept being enabled and it is not as harmless as it is presumed in the topic describing animations where it states - "Disabling animations will not improve performance drastically". Disabling animation drastically improved performance in my case.
   I have 3 charts with about 4000 points each and I display only a portion of the charts specifying MinLimit and MaxLimit and then I have to slide them left and right in sync with my mouse. If I slide only one chart, I have to admit the performance is acceptable but even in this case the CPU is way overloaded with calculation. There is MotionProperty.GetMovement method. I didn't get too deep into the details, but this method eats up 34% CPU out of total 61% related to animation when the animation is enabled when I slide the charts left and right. Then, without being able to disable animation gracefully I commented out all 44 occurrences of

`        .WithAnimation(animation =>
            animation
                .WithDuration(delayedFunction.Speed)
                .WithEasingFunction(delayedFunction.Function))
`

and after that everything was flying. No delays when I move the chart whatsoever. All three charts redraw instantly and in sync as I wanted. Do you think you can address that issue which I would classify as a serious one and allow global disabling of animation?

Thank you

**Reply by alex4998** [reply]
Any thoughts about this?

**Reply by alex4998** [reply]
Anything? Any hope to be able to turn off the animation?

**Reply by alex4998** [reply]
I still hope this will get proper attention.

**Reply by beto-rodriguez** [reply]
which platform?

**Reply by mjanulaitis** [reply]
Windows

**Reply by mjanulaitis** [reply]
@beto-rodriguez sorry I missed your comment.  'EasingFunction = null' works perfect.
